DQM/L1TMonitor/src/L1TObjectsTiming.cc

```diff
@@ -0,0 +1,480 @@
+    }
+    if (!gtUtil_->getAlgBitFromName(algoNameLastBxInTrain_, algoBitLastBxInTrain_)) {
+      edm::LogWarning("L1TObjectsTiming") << "Algo \"" << algoNameLastBxInTrain_ << "\" not found in the trigger menu " << gtUtil_->gtTriggerMenuName() << ". Could not retrieve algo bit number.";
+    }
+
+
+}
+
+void L1TObjectsTiming::beginLuminosityBlock(const edm::LuminosityBlock&, const edm::EventSetup&) {}
+
+void L1TObjectsTiming::bookHistograms(DQMStore::IBooker& ibooker, const edm::Run&, const edm::EventSetup&) {
+
+
+  // Subsystem Monitoring and Muon Output
+  std::string bx_obj[5]={"minus2","minus1","0","plus1","plus2"};
+
+  ibooker.setCurrentFolder(monitorDir+"/L1TMuon"+"/timing"); 
+  for(unsigned int i=0; i<bxrange; ++i) {
+    muons_eta_phi.push_back(ibooker.book2D("muons_eta_phi_bx_"+bx_obj[i],"L1T Muon #eta vs #phi BX="+bx_obj[i],25, -2.5, 2.5, 25, -3.2, 3.2));
+    muons_eta_phi.at(i)->setAxisTitle("#eta", 1);
+    muons_eta_phi.at(i)->setAxisTitle("#phi", 2);
+  }
+  
+  ibooker.setCurrentFolder(monitorDir+"/L1TJet"+"/timing"); 
+  for(unsigned int i=0; i<bxrange; ++i) { 
+    jet_eta_phi.push_back(ibooker.book2D("jet_eta_phi_bx_"+bx_obj[i],"L1T Jet #eta vs #phi BX="+bx_obj[i],25, -2.5, 2.5, 25, -3.2, 3.2));
+    jet_eta_phi.at(i)->setAxisTitle("#eta", 1);
+    jet_eta_phi.at(i)->setAxisTitle("#phi", 2);
+  }
+   
+  ibooker.setCurrentFolder(monitorDir+"/L1TEGamma"+"/timing"); 
+  for(unsigned int i=0; i<bxrange; ++i) {
+    egamma_eta_phi.push_back(ibooker.book2D("egamma_eta_phi_bx_"+bx_obj[i],"L1T EGamma #eta vs #phi BX="+bx_obj[i],25, -2.5, 2.5, 25, -3.2, 3.2));
+    egamma_eta_phi.at(i)->setAxisTitle("#eta", 1);
+    egamma_eta_phi.at(i)->setAxisTitle("#phi", 2);
+  }
+
+  ibooker.setCurrentFolder(monitorDir+"/L1TTau"+"/timing");
+  for(unsigned int i=0; i<bxrange; ++i) {
+    tau_eta_phi.push_back(ibooker.book2D("tau_eta_phi_bx_"+bx_obj[i],"L1T Tau #eta vs #phi BX="+bx_obj[i],25, -2.5, 2.5, 25, -3.2, 3.2));
+    tau_eta_phi.at(i)->setAxisTitle("#eta", 1);
+    tau_eta_phi.at(i)->setAxisTitle("#phi", 2);
+  }
+
+  ibooker.setCurrentFolder(monitorDir+"/L1TEtSum"+"/timing");
+  for(unsigned int i=0; i<bxrange; ++i) {
+    etsum_eta_phi_MET.push_back(ibooker.book1D("etsum_phi_bx_MET_"+bx_obj[i],"L1T MET #phi BX="+bx_obj[i],25, -3.2, 3.2));
+    etsum_eta_phi_MET.at(i)->setAxisTitle("#phi", 1);
+    etsum_eta_phi_METHF.push_back(ibooker.book1D("etsum_phi_bx_METHF_"+bx_obj[i],"L1T METHF #phi BX="+bx_obj[i],25, -3.2, 3.2));
+    etsum_eta_phi_METHF.at(i)->setAxisTitle("#phi", 1);
+    etsum_eta_phi_MHT.push_back(ibooker.book1D("etsum_phi_bx_MHT_"+bx_obj[i],"L1T MHT #phi BX="+bx_obj[i],25, -3.2, 3.2));
+    etsum_eta_phi_MHT.at(i)->setAxisTitle("#phi", 1);
+    etsum_eta_phi_MHTHF.push_back(ibooker.book1D("etsum_phi_bx_MHTHF_"+bx_obj[i],"L1T MHTHF #phi BX="+bx_obj[i],25, -3.2, 3.2));
+    etsum_eta_phi_MHTHF.at(i)->setAxisTitle("#phi", 1);
+  }
+
+
```

Comment:
Only create and fill these MEs if isolated bunches exist. So `algoBitFirstBxInTrain_ > -1 && algoBitLastBxInTrain_ > -1`